ministryofjustice/peoplefinder
# CT-2520 Remove Omniauth/Google login

## Description
Further updates to gem dependencies - removal of OmniAuth owing to https://nvd.nist.gov/vuln/detail/CVE-2015-9284 

This was previously used for option to use Google login instead of emailing yourself a link to log in. This no longer works; a known problem arising because Google withdrew support for the mechanism they used - part of them sunsetting Google+ 

Because the feature doesn't work any more, I believe we are not exposed to the risk, but the automated security check for dependency vulnerabilities on github has just flagged it again and just to be on the safe side I propose to remove the Google sign in feature 

So this PR removes all aspects of the OAuth sign in and associated tests.

> CVE-2015-9284
> high severity
> Vulnerable versions: <= 1.9.0
> Patched version: No fix
> The request phase of the OmniAuth Ruby gem is vulnerable to Cross-Site Request Forgery when used as part of the Ruby on Rails framework, allowing accounts to be connected without user intent, user interaction, or feedback to the user. This permits a secondary account to be able to sign into the web application as the primary account.

And HAML bumped to 5.0.0 for https://nvd.nist.gov/vuln/detail/CVE-2017-1002201

PLUS

Update the date of the survey to two weeks from tomorrow

 
## Self-review checklist

 
### Screenshots
BEFORE 

![image](https://user-images.githubusercontent.com/1161161/67529619-94a68d00-f6b4-11e9-9c3e-faea94c6dca3.png)

AFTER

![image](https://user-images.githubusercontent.com/1161161/67529598-848ead80-f6b4-11e9-86ae-a7cf064c8bf0.png)

 
### Related JIRA tickets
https://dsdmoj.atlassian.net/browse/CT-2572
https://dsdmoj.atlassian.net/browse/CT-2520

 
### Deployment
Nothing new to report

### Manual testing instructions
Use the app - view the banner where it appears on home page, search results page, and Person profile page 

Confirm the Google login no longer appears as an option on the sign in page
